app/src/main/java/pl/droidevs/books/library/LibraryActivity.java

```diff
@@ -190,21 +192,36 @@ public void onSwiped(int position) {
     }
 
     private void showRemoveBookSnackbar(@NonNull Book book, int position) {
-        Snackbar.make(floatingActionButton, R.string.book_delete_snackbar, Snackbar.LENGTH_LONG)
+        Snackbar.make(floatingActionButton, R.string.removing_book_snackbar, Snackbar.LENGTH_LONG)
                 .setAction(R.string.undo, v -> adapter.addItem(book, position))
                 .addCallback(
                         new BaseTransientBottomBar.BaseCallback<Snackbar>() {
                             @Override
                             public void onDismissed(Snackbar transientBottomBar, int event) {
                                 super.onDismissed(transientBottomBar, event);
-                                libraryViewModel.removeBook(book);
+
+                                removeBook(book);
                             }
                         })
                 .show();
     }
 
+    private void removeBook(Book book) {
+        RemoveBookViewModel removeBookViewModel = ViewModelProviders
```

Comment:
wydaje mi się, że korzystanie z ViewModelu innego widoku tylko dlatego że ma potrzebną logikę jest antywzorcem, raczej ta logika powinna być przesunięta do osobnej klasy z której korzystają obydwa ViewModele

Reply:
hmm... nie zgodzę się. Pierwszą rzeczą jaką zapamiętałam i jednocześnie mi się spodobała w MVVM było to, że w przeciwieństwie do MVP możesz mieć więcej ViewModeli per widok.
https://medium.com/@Miqubel/refactoring-to-mvvm-40ebafff43de

I też nie ma widoku usuwania, jest tylko viewmodel który jest użyty w LibraryActivity oraz w SaveActivity

Reply:
ok troche poczytałem i faktycznie tak jest, muszę się przestawić z przyzwyczajeń MVP